modules/files.nix

```diff
@@ -260,7 +265,15 @@ in
 
           if [[ ! -v oldGenPath || "$oldGenPath" != "$newGenPath" ]] ; then
             echo "Creating profile generation $newGenNum"
-            $DRY_RUN_CMD nix-env $VERBOSE_ARG --profile "$genProfilePath" --set "$newGenPath"
+            if [[ -e "$genProfilePath"/manifest.json ]] ; then
+              # Remove all packages from "$genProfilePath"
+              # `nix profile remove '.*' --profile "$genProfilePath"` was not working, so here is a workaround:
+              nix profile list --profile "$genProfilePath" | awk -F ' ' '{ print $4 }' | xargs -t $DRY_RUN_CMD nix profile remove $VERBOSE_ARG --profile "$genProfilePath"
+            else
+              $DRY_RUN_CMD nix-env $VERBOSE_ARG --profile "$genProfilePath" --set "$newGenPath"
+            fi
+
+            $DRY_RUN_CMD nix profile install $VERBOSE_ARG --profile "$genProfilePath" "$newGenPath"
```

Comment:
- `nix profile` expands to nix `profile --profile /nix/var/nix/profiles/per-user/$user`
- `"$genProfilePath"/manifest.json` expands to `/nix/var/nix/profiles/per-user/$user/home-manager/manifest.json`

Which one is correct?

---

It also strikes me as volatile to rely on the environment to determine the profile path, in case the environment user doesn't match the intended target user. Such env user could be a deployment user.